eb_hooks.py

```diff
@@ -612,39 +633,37 @@ def pre_fetch_hook_zen4_gcccore1220(self, *args, **kwargs):
         print_msg("Updated build option 'force' to 'True'")
 
 
-def pre_module_hook_zen4_gcccore1220(self, *args, **kwargs):
+def pre_module_hook_unsupported_module(self, *args, **kwargs):
     """Make module load-able during module step"""
-    if is_gcccore_1220_based(ecname=self.name, ecversion=self.version, tcname=self.toolchain.name,
-                             tcversion=self.toolchain.version):
-        if hasattr(self, 'initial_environ'):
-            # Allow the module to be loaded in the module step (which uses initial environment)
-            print_msg(f"Setting {EESSI_IGNORE_ZEN4_GCC1220_ENVVAR} in initial environment")
-            self.initial_environ[EESSI_IGNORE_ZEN4_GCC1220_ENVVAR] = "1"
-
-
-def post_module_hook_zen4_gcccore1220(self, *args, **kwargs):
-    """Revert changes from pre_fetch_hook_zen4_gcccore1220"""
-    if is_gcccore_1220_based(ecname=self.name, ecversion=self.version, tcname=self.toolchain.name,
-                             tcversion=self.toolchain.version):
-        if hasattr(self, EESSI_MODULE_ONLY_ATTR):
-            update_build_option('module_only', getattr(self, EESSI_MODULE_ONLY_ATTR))
-            print_msg("Restored original build option 'module_only' to %s" % getattr(self, EESSI_MODULE_ONLY_ATTR))
-        else:
-            raise EasyBuildError("Cannot restore module_only to it's original value: 'self' is missing attribute %s.",
-                                 EESSI_MODULE_ONLY_ATTR)
+    ignore_lmoderror_envvar = is_unsupported_module(self)
+    if ignore_lmoderror_envvar and hasattr(self, 'initial_environ'):
+        # Allow the module to be loaded in the module step (which uses initial environment)
+        print_msg(f"Setting {ignore_lmoderror_envvar} in initial environment")
+        self.initial_environ[ignore_lmoderror_envvar] = "1"
+
+
+def post_module_hook_unsupported_module(self, *args, **kwargs):
+    """Revert changes from pre_fetch_hook_unsupported_modules"""
+    ignore_lmoderror_envvar = is_unsupported_module(self)
+    if hasattr(self, EESSI_MODULE_ONLY_ATTR):
```

Comment:
Doesn't everything below need to be protected by an
```
if ignore_lmoderror_envvar:
```
otherwise every easyconfig that does not have the the attribute is going to error out

Reply:
I see now that you only end up there if that is true, but I still think it is worth including. You may want to consider deleting the attribute as well, so there can be a generic check to ensure that these only exist when we expect them to exist.

Reply:
It's protected by such an if statement in the general post_module_hook here:
https://github.com/EESSI/software-layer-scripts/pull/91/files#diff-03a1681f29b92037b3b19f52498d8a045fb14dcdebbaaadee3ffaaaa2dcd2ea7R1565
But I can move it to the hook itself to make that more explicit and be sure that it's always checked.